dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/condition/ConditionRouter.java

```diff
@@ -262,6 +276,54 @@ private boolean matchCondition(Map<String, MatchPair> condition, URL url, URL pa
         return result;
     }
 
+    /**
+     * analysis the arguments in the rule.
+     * Examples would be like this:
+     * "arguments[0]=1", whenCondition is that the first argument is equal to '1'.
+     * "arguments[1].param=a", whenCondition is that the 'param' attribute in the second argument is equal to 'a'.
+     * "arguments[2].inner.param=b", whenCondition is that the 'param' attribute in the 'inner' attribute of the third argument is equal to 'b'.

```

Comment:
I think it's a good complement to have an argument matcher.

I think we can make the argument routing rules simpler while still covering more than 90% of the usage scenarios.

For example, we only support the matching of string type parameters and do not support t matching of complex objects like `arguments[1].param`. Or even our routing rule is like a switch, when the user tells us to turn it on, we will constantly match the first parameter of the method and require that this parameter be a string or primitive type.

Reply:
> I think it's a good complement to have an argument matcher.
> 
> I think it's better if we can make the argument routing rules simpler while still covering more than 90% of the usage scenarios.
> 
> For example, we only support the matching of string type parameters and do not support t matching of complex objects like `arguments[1].param`. Or even our routing rule is like a switch, when the user tells us to turn it on, we will constantly match the first parameter of the method and require that this parameter be a string or primitive type.

I would like to simplify the arguments rule to make it easier to be understood and used.
For example,  we only support "one flour param", like  "arguments[0]=1", and only support java primary data type, like int, long, string and so on.
But I think it is not a good idea to make it like a switch, that would reduce flexibility. 
And in the future, I would like to config arguments router in the configuration pages, we could use drop-down box to choose the right arguments and its value. And in that time, we could expand the explementation of arguments router.
@AlbumenJ